sim/mage/combustion.go

```diff
@@ -0,0 +1,139 @@
+
+						}
+					}
+					combustionDotDamage += normalizedDPS
+				} */
+				var spellDPS float64
+				for _, spell := range dotSpells {
+					dots := spell.Dot(mage.CurrentTarget)
+					// EJ states that combustion double dips on mastery for LB + Pyro, but not ignite
+					// https://web.archive.org/web/20120207223126/http://elitistjerks.com/f75/t110187-cataclysm_mage_simulators_formulators/p3/
+					if spell != mage.Ignite {
+						if dots != nil && dots.IsActive() {
+							spellDPS = spell.Dot(mage.CurrentTarget).CalcSnapshotDamage(sim, mage.CurrentTarget, dots.OutcomeTick).Damage / 3
+							fmt.Println(dots.Label, " snapped for : ", spellDPS)
+						}
+					} else {
+						//This part is for ignite. The denominator will probably be variable if it works as intended in cata.
+						if dots != nil && dots.IsActive() {
+							spellDPS = spell.Dot(mage.CurrentTarget).SnapshotBaseDamage / 2
+							fmt.Println(dots.Label, " snapped for : ", spellDPS)
+						}
+					}
+					combustionDotDamage += spellDPS
+				}
+				dot.Snapshot(mage.CurrentTarget, combustionDotDamage)
+			},
+			OnTick: func(sim *core.Simulation, target *core.Unit, dot *core.Dot) {
+				dot.CalcAndDealPeriodicSnapshotDamage(sim, target, dot.OutcomeSnapshotCrit)
+			},
+		},
+		DamageMultiplierAdditive: 1,
+		CritMultiplier:           mage.DefaultSpellCritMultiplier(),
+		BonusCoefficient:         1.113,
+		ThreatMultiplier:         1,
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			baseDamage := 0.429 * mage.ScalingBaseDamage
+			result := spell.CalcDamage(sim, target, baseDamage, spell.OutcomeMagicHit)
+			spell.DealDamage(sim, result)
+			spell.Dot(target).Apply(sim)
+		},
+	})
+
+	mage.AddMajorCooldown(core.MajorCooldown{
+		Spell: mage.Combustion,
+		Type:  core.CooldownTypeDPS,
+	})
+
+	mage.CombustionImpact = mage.RegisterSpell(core.SpellConfig{
+		ActionID:       core.ActionID{SpellID: 11129}.WithTag(1),
+		SpellSchool:    core.SpellSchoolFire,
+		ProcMask:       core.ProcMaskSpellDamage,
+		Flags:          core.SpellFlag(MageSpellFlagNone),
+		ClassSpellMask: MageSpellCombustion,
+
+		Cast: core.CastConfig{
+			DefaultCast: core.Cast{
+				GCD: core.GCDDefault,
+			},
+		},
+
+		Dot: core.DotConfig{
+			Aura: core.Aura{
+				Label: "Combustion Impact Fake Dot",
+			},
+			NumberOfTicks:       10,
+			TickLength:          time.Second,
+			AffectedByCastSpeed: true,
+			OnSnapshot: func(sim *core.Simulation, target *core.Unit, dot *core.Dot, _ bool) {
+
+			},
+			OnTick: func(sim *core.Simulation, target *core.Unit, dot *core.Dot) {
+				dot.CalcAndDealPeriodicSnapshotDamage(sim, target, dot.OutcomeSnapshotCrit)
+			},
+		},
+		DamageMultiplierAdditive: 1,
+		CritMultiplier:           mage.DefaultSpellCritMultiplier(),
+		ThreatMultiplier:         1,
+
+		ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+			spell.Dot(target).ApplyOrReset(sim)
```

Comment:
If combustion can miss you should check here after you move the impact damage. (It looks like it has a hit check from your code)

Reply:
Just asked and mage discord says combustion can miss